src/Symfony/Component/RateLimiter/Rate.php

```diff
@@ -0,0 +1,95 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\RateLimiter;
+
+/**
+ * Data object representing the fill rate of a token bucket.
+ *
+ * @author Wouter de Jong <wouter@wouterj.nl>
+ *
+ * @final
+ */
+class Rate
+{
+    const UNIT_HOUR = 'h';
+    const UNIT_MINUTE = 'm';
+    const UNIT_SECOND = 's';
+
+    private $perSecond;
+    private $initialUnit;
+
+    private function __construct(float $ratePerSecond, string $initialUnit)
+    {
+        $this->perSecond = $ratePerSecond;
+        $this->initialUnit = $initialUnit;
+    }
+
+    public static function perSecond(int $rate): self
+    {
+        return new static($rate, self::UNIT_SECOND);
+    }
+
+    public static function perMinute(int $rate): self
+    {
+        return new static($rate / 60, self::UNIT_MINUTE);
+    }
+
+    public static function perHour(int $rate): self
+    {
+        return new static($rate / 3600, self::UNIT_HOUR);
```

Comment:
When I want a Rate of 3600 per hour, it's not the same as 1 per second. In the first case, consuming 3600 tokens right away should be possible, and then you cannot consume anything anymore for about an hour minus 1 second.

I might have missed something, but it looks like here you are enforcing 1 per second.

Reply:
Agreed, this is a missing feature. The Go package I used as a reference has this limitation. We should keep the original unit and split "fill time" and "fill amount" as described in https://medium.com/smyte/rate-limiter-df3408325846